manual/source/classes/contributed_definitions/ellipsometry-structure.rst

```diff
@@ -0,0 +1,179 @@
+       adding a dielectric background (e.g. Sellmeier model) to an oscillator model (e.g. Lorentz).
+              
+    :ref:`NXdispersion_function`
+       This dispersion is described by a function and its single and repeated parameter values.
+       It follows a formula of the form ``eps = eps_inf + sum[A * lambda ** 2 / (lambda ** 2 - B ** 2)]`` 
+       (Sellmeier formula). See the formula grammar below for an ebnf grammar for this form.
+
+    :ref:`NXdispersion_single_parameter`
+       This denotes a parameter which is used outside the summed part of a dispersion function,
+       e.g. ``eps_inf`` in the formula example above.
+
+    :ref:`NXdispersion_repeated_parameter`
+       This denotes arrays of repeated parameters which are used to build a sum of parameter values, e.g.
+       ``A`` and ``B`` are repeated parameters in the formula above.
+       
+    :ref:`NXdispersion_table`
+       This describes a tabular dispersion where the dielectric function is an array versus wavelength or energy.
+
+Formula Grammar
+---------------
+
+Below you find a grammar to which the formula should adhere and which can be used to parse and
+evaluate the dispersion function. The terms ``single_param_name`` and ``param_name`` should be
+filled with the respective single and repeated params from the stored data.
+
+.. code-block::
+
+   ?assignment: "eps" "=" kkr_expression -> eps
+             | "n" "=" kkr_expression -> n
+
+   ?kkr_expression: expression
+                  | "<kkr>" "+" "1j" "*" term -> kkr_term
+
+   ?expression: term
+               | expression "+" term -> add
+               | expression "-" term -> sub
+
+   ?term: factor
+         | term "*" factor -> mul
+         | term "/" factor -> div
+
+   ?factor: power
+         | power "**" power -> power
+
+
+   ?power: "(" expression ")"
+         | FUNC "(" expression ")" -> func
+         | "sum" "[" repeated_expression "]" -> sum_expr
+         | NAME -> single_param_name
+         | SIGNED_NUMBER -> number
+         | BUILTIN -> builtin
+
+   ?repeated_expression: repeated_term
+                     | repeated_expression "+" repeated_term -> add
+                     | repeated_expression "-" repeated_term -> sub
+
+
+   ?repeated_term: repeated_factor
+                  | repeated_term "*" repeated_factor -> mul
+                  | repeated_term "/" repeated_factor -> div
+
+   ?repeated_factor: repeated_power
+                     | repeated_power "**" repeated_power -> power
+
+   ?repeated_power: "(" repeated_expression ")"
+                  | FUNC "(" repeated_expression ")" -> func
+                  | SIGNED_NUMBER -> number
+                  | NAME -> param_name
+                  | BUILTIN -> builtin
+
+   FUNC.1: "sin" | "cos" | "tan" | "sqrt" | "dawsn" | "ln" | "log" | "heaviside" 
+   BUILTIN.1: "1j" | "pi" | "eps_0" | "hbar" | "h" | "c" 
+
+   %import common.CNAME -> NAME
```

Comment:
This  seems to imply this example is incomplete

Reply:
The `%import` statement. The grammar is also unspecified.

Reply:
Yes, this grammar is basically [lark](https://github.com/lark-parser/lark) syntax, which is very close to ebnf. I essentially copy and pasted it here to describe what I build for this and bring it towards a standardised way of building such formulas.

[CNAME](https://github.com/lark-parser/lark/blob/942366b49247e996e387cb901ed96c7d861382a0/lark/grammars/common.lark#L41C40-L41C40) is really just describing a common name, i.e., something which would be a valid python variable name. So I didn't bother to copy it here, because I wasn't really sure whether this makes sense to have the grammer even here. It should, however, be easy to make this a complete and proper ebnf grammar.

Reply:
I think this description of the DSL should be standalone and as standard as possible. If it can be expressed as an EBNF grammar then do so or if you can only do so using lark then reference that.